minecraft_go/software/mc_go/include/gps.h

```diff
@@ -20,21 +27,66 @@
 #define GPS_DATA_DUMP_PARTIAL "$PMTK622,1*29\r\n"
 #define GPS_DATA_DUMP_END "$PMTKLOX,2*47\r\n"
 
-#define GGA_IDENTIFIER "$GPGGA"
-
-struct gga_data {
+typedef struct GGA_data {
 	char UTC_time[11];
 	char latitude[10];
 	char N_S[2];
 	char longitude[11];
 	char E_W[2];
 	char satellites[3];
-};
+} GGA_data;
+
+typedef struct RMC_data {
+	char UTC_time[11]; // hhmmss.sss
+	char latitude[10]; // ddmm.mmmm
+	char N_S[2];
+	char longitude[11]; // dddmm.mmmm
+	char E_W[2];
+	char speed[5]; // knots
+	char date[7]; //ddmmyy
+} RMC_data;
 
-char * gps_retrive_data_line(int);
+typedef struct DateTime {
+	int year;
+	int month;
+	int day;
+	int hour;
+	int minute;
+	int second;
+} DateTime;
+
+typedef struct Time {
+	unsigned long hour;
+	int minute;
+	int second;
+} Time;
+
+typedef struct Location {
+	int Lat_degree;
```

Comment:
style nitpick: is there a reason why these fields are capitalized when the ones above are not?